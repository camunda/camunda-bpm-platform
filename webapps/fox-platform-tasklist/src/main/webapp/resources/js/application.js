// NOTICE!! DO NOT USE ANY OF THIS JAVASCRIPT
// IT'S ALL JUST JUNK FOR OUR DOCS!
// ++++++++++++++++++++++++++++++++++++++++++

!function ($) {

  $(function(){

    // Disable certain links in docs
    $('section [href^=#]').click(function (e) {
      e.preventDefault()
    })

    // make code pretty
    window.prettyPrint && prettyPrint()

    // add-ons
    $('.add-on :checkbox').on('click', function () {
      var $this = $(this)
        , method = $this.attr('checked') ? 'addClass' : 'removeClass'
      $(this).parents('.add-on')[method]('active')
    })

    // position static twipsies for components page
    if ($(".twipsies a").length) {
      $(window).on('load resize', function () {
        $(".twipsies a").each(function () {
          $(this)
            .tooltip({
              placement: $(this).attr('title')
            , trigger: 'manual'
            })
            .tooltip('show')
          })
      })
    }

    // add tipsies to grid for scaffolding
    if ($('#grid-system').length) {
      $('#grid-system').tooltip({
          selector: '.show-grid > div'
        , title: function () { return $(this).width() + 'px' }
      })
    }

    // fix sub nav on scroll
    var $win = $(window)
      , $nav = $('.subnav')
      , navTop = $('.subnav').length && $('.subnav').offset().top - 40
      , isFixed = 0

    processScroll()

    // hack sad times - holdover until rewrite for 2.1
    $nav.on('click', function () {
      if (!isFixed) setTimeout(function () {  $win.scrollTop($win.scrollTop() - 47) }, 10)
    })

    $win.on('scroll', processScroll)

    function processScroll() {
      var i, scrollTop = $win.scrollTop()
      if (scrollTop >= navTop && !isFixed) {
        isFixed = 1
        $nav.addClass('subnav-fixed')
      } else if (scrollTop <= navTop && isFixed) {
        isFixed = 0
        $nav.removeClass('subnav-fixed')
      }
    }

    // tooltip demo
    $('.tooltip-demo.well').tooltip({
      selector: "a[rel=tooltip]"
    })

    $('.tooltip-test').tooltip()
    $('.popover-test').popover()

    // popover demo
    $("a[rel=popover]")
      .popover()
      .click(function(e) {
        e.preventDefault()
      })

    // button state demo
    $('#fat-btn')
      .click(function () {
        var btn = $(this)
        btn.button('loading')
        setTimeout(function () {
          btn.button('reset')
        }, 3000)
      })

    // carousel demo
    $('#myCarousel').carousel()

    // javascript build logic
    var inputsComponent = $("#components.download input")
      , inputsPlugin = $("#plugins.download input")
      , inputsVariables = $("#variables.download input")

    // toggle all plugin checkboxes
    $('#components.download .toggle-all').on('click', function (e) {
      e.preventDefault()
      inputsComponent.attr('checked', !inputsComponent.is(':checked'))
    })

    $('#plugins.download .toggle-all').on('click', function (e) {
      e.preventDefault()
      inputsPlugin.attr('checked', !inputsPlugin.is(':checked'))
    })

    $('#variables.download .toggle-all').on('click', function (e) {
      e.preventDefault()
      inputsVariables.val('')
    })

    // request built javascript
    $('.download-btn').on('click', function () {

      var css = $("#components.download input:checked")
            .map(function () { return this.value })
            .toArray()
        , js = $("#plugins.download input:checked")
            .map(function () { return this.value })
            .toArray()
        , vars = {}
        , img = ['glyphicons-halflings.png', 'glyphicons-halflings-white.png']

    $("#variables.download input")
      .each(function () {
        $(this).val() && (vars[ $(this).prev().text() ] = $(this).val())
      })

      $.ajax({
        type: 'POST'
      , url: /\?dev/.test(window.location) ? 'http://localhost:3000' : 'http://bootstrap.herokuapp.com'
      , dataType: 'jsonpi'
      , params: {
          js: js
        , css: css
        , vars: vars
        , img: img
      }
      })
    })
  })

// Modified from the original jsonpi https://github.com/benvinegar/jquery-jsonpi
$.ajaxTransport('jsonpi', function(opts, originalOptions, jqXHR) {
  var url = opts.url;

  return {
    send: function(_, completeCallback) {
      var name = 'jQuery_iframe_' + jQuery.now()
        , iframe, form

      iframe = $('<iframe>')
        .attr('name', name)
        .appendTo('head')

      form = $('<form>')
        .attr('method', opts.type) // GET or POST
        .attr('action', url)
        .attr('target', name)

      $.each(opts.params, function(k, v) {

        $('<input>')
          .attr('type', 'hidden')
          .attr('name', k)
          .attr('value', typeof v == 'string' ? v : JSON.stringify(v))
          .appendTo(form)
      })

      form.appendTo('body').submit()
    }
  }
})

}(window.jQuery)