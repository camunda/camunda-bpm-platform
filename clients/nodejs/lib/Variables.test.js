/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information regarding copyright
 * ownership. Camunda licenses this file to you under the Apache License,
 * Version 2.0; you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Variables, File } from "../index.js";

describe("Variables", () => {
  describe("read-only", () => {
    it("should only have getters if readOnly is true", () => {
      // given
      const readOnlyVariables = new Variables({}, { readOnly: true });

      // then
      expect(Object.keys(readOnlyVariables)).toMatchSnapshot();
    });

    it("should have getters and setters if readOnly is not true", () => {
      // given
      const readOnlyVariables = new Variables({});

      // then
      expect(Object.keys(readOnlyVariables)).toMatchSnapshot();
    });
  });

  describe("getters", () => {
    let variables;
    beforeEach(() => {
      variables = new Variables({
        foo: { type: "string", value: "FooValue", valueInfo: {} },
        bar: { type: "integer", value: 2, valueInfo: {} },
        baz: { type: "json", value: '{"name":"baz"}', valueInfo: {} },
        qux: {
          type: "date",
          value: new Date("2018-01-23T14:42:45.435+0200"),
          valueInfo: {},
        },
        zex: {
          type: "file",
          value: null,
          valueInfo: {},
        },
      });

      const file = new File({ localPath: "some/local/path" });
      file.content = Buffer.from("some content");

      variables.setTyped("blax", {
        type: "file",
        value: file,
        valueInfo: {},
      });
    });

    it("getAllTyped() should return all variables", () => {
      expect(variables.getAllTyped()).toMatchSnapshot();
    });

    it("getAll() should return values of all variables", () => {
      expect(variables.getAll()).toMatchSnapshot();
    });

    it("getDirtyVariables() should return all dirty variables", () => {
      expect(variables.getDirtyVariables()).toMatchSnapshot();
    });

    it("get('foo') should return value of key foo", () => {
      expect(variables.get("foo")).toMatchSnapshot();
    });

    it("getTyped('non_existing_key') should return null", () => {
      expect(variables.getTyped("non_existing_key")).toBeNull();
    });

    it("getTyped('foo') should return the typed value of key foo", () => {
      expect(variables.getTyped("foo")).toMatchSnapshot();
    });
  });

  describe("setters", () => {
    let variables;
    beforeEach(() => {
      variables = new Variables();
    });

    it('setTyped("baz",someTypeValue) should set typed value with key "baz"', () => {
      // given
      expect(variables.getAllTyped()).toMatchSnapshot("variables");
      expect(variables.getDirtyVariables()).toMatchSnapshot("dirty variables");
      const key = "baz";
      const typedValue = {
        type: "json",
        value: { name: "bazname" },
        valueInfo: {},
      };

      // when
      variables.setTyped(key, typedValue);

      // then
      expect(variables.getAllTyped()).toMatchSnapshot("variables");
      expect(variables.getDirtyVariables()).toMatchSnapshot("dirty variables");
    });

    it("setAllTyped(someTypedValues) should add someTypedValues to variables", () => {
      // given
      expect(variables.getAllTyped()).toMatchSnapshot("variables");
      expect(variables.getDirtyVariables()).toMatchSnapshot("dirty variables");
      const typedValues = {
        foo: { value: "fooValue", type: "string", valueInfo: {} },
      };
      variables.set("bar", "barValue");

      // when
      variables.setAllTyped(typedValues);

      // then
      expect(variables.getAllTyped()).toMatchSnapshot("variables");
      expect(variables.getDirtyVariables()).toMatchSnapshot("dirty variables");
    });

    it('set("foo", "fooValue")) should set variable with key "foo" and value "fooValue"', () => {
      // given
      expect(variables.getAllTyped()).toMatchSnapshot("variables");
      expect(variables.getDirtyVariables()).toMatchSnapshot("dirty variables");
      variables.set("foo", "fooValue");

      // then
      expect(variables.getAllTyped()).toMatchSnapshot("variables");
      expect(variables.getDirtyVariables()).toMatchSnapshot("dirty variables");
    });

    it('setTransient("fooTransient", "fooValueTransient")) should set variable with key "fooTransient" and value "fooValueTransient" and transient "true"', () => {
      // given
      expect(variables.getAllTyped()).toMatchSnapshot("variables");
      expect(variables.getDirtyVariables()).toMatchSnapshot("dirty variables");
      variables.setTransient("fooTransient", "fooValueTransient");

      // then
      expect(variables.getAllTyped()).toMatchSnapshot("variables");
      expect(variables.getDirtyVariables()).toMatchSnapshot("dirty variables");
    });

    it("setAll(someValues)  should add someValues to variables", () => {
      // given
      expect(variables.getAllTyped()).toMatchSnapshot("variables");
      expect(variables.getDirtyVariables()).toMatchSnapshot("dirty variables");
      const someValues = {
        foo: "FooValue",
        bar: 2,
      };

      // when
      variables.setAll(someValues);

      // then
      // given
      expect(variables.getAllTyped()).toMatchSnapshot("variables");
      expect(variables.getDirtyVariables()).toMatchSnapshot("dirty variables");
    });
  });
});
