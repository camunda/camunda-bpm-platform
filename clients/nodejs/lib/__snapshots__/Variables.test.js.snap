// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Variables getters get('foo') should return value of key foo 1`] = `"FooValue"`;

exports[`Variables getters getAll() should return values of all variables 1`] = `
{
  "bar": 2,
  "baz": {
    "name": "baz",
  },
  "blax": File {
    "__readFile": [Function],
    "content": "",
    "createTypedValue": [Function],
    "engineService": undefined,
    "filename": "path",
    "load": [Function],
    "remotePath": "/execution/undefined/localVariables/blax/data",
  },
  "foo": "FooValue",
  "qux": 2018-01-23T12:42:45.435Z,
  "zex": File {
    "__readFile": [Function],
    "content": "",
    "createTypedValue": [Function],
    "engineService": undefined,
    "filename": "data",
    "load": [Function],
    "remotePath": "/execution/undefined/localVariables/zex/data",
  },
}
`;

exports[`Variables getters getAllTyped() should return all variables 1`] = `
{
  "bar": {
    "type": "integer",
    "value": 2,
    "valueInfo": {},
  },
  "baz": {
    "type": "json",
    "value": {
      "name": "baz",
    },
    "valueInfo": {},
  },
  "blax": {
    "type": "file",
    "value": File {
      "__readFile": [Function],
      "content": "",
      "createTypedValue": [Function],
      "engineService": undefined,
      "filename": "path",
      "load": [Function],
      "remotePath": "/execution/undefined/localVariables/blax/data",
    },
    "valueInfo": {
      "filename": "path",
    },
  },
  "foo": {
    "type": "string",
    "value": "FooValue",
    "valueInfo": {},
  },
  "qux": {
    "type": "date",
    "value": 2018-01-23T12:42:45.435Z,
    "valueInfo": {},
  },
  "zex": {
    "type": "file",
    "value": File {
      "__readFile": [Function],
      "content": "",
      "createTypedValue": [Function],
      "engineService": undefined,
      "filename": "data",
      "load": [Function],
      "remotePath": "/execution/undefined/localVariables/zex/data",
    },
    "valueInfo": {},
  },
}
`;

exports[`Variables getters getDirtyVariables() should return all dirty variables 1`] = `
{
  "blax": {
    "type": "file",
    "value": "c29tZSBjb250ZW50",
    "valueInfo": {
      "filename": "path",
    },
  },
}
`;

exports[`Variables getters getTyped('foo') should return the typed value of key foo 1`] = `
{
  "type": "string",
  "value": "FooValue",
  "valueInfo": {},
}
`;

exports[`Variables read-only should have getters and setters if readOnly is not true 1`] = `
[
  "getTyped",
  "get",
  "getAll",
  "getAllTyped",
  "getDirtyVariables",
  "setTyped",
  "set",
  "setTransient",
  "setAll",
  "setAllTyped",
]
`;

exports[`Variables read-only should only have getters if readOnly is true 1`] = `
[
  "getTyped",
  "get",
  "getAll",
  "getAllTyped",
  "getDirtyVariables",
]
`;

exports[`Variables setters set("foo", "fooValue")) should set variable with key "foo" and value "fooValue": dirty variables 1`] = `{}`;

exports[`Variables setters set("foo", "fooValue")) should set variable with key "foo" and value "fooValue": dirty variables 2`] = `
{
  "foo": {
    "type": "string",
    "value": "fooValue",
    "valueInfo": {},
  },
}
`;

exports[`Variables setters set("foo", "fooValue")) should set variable with key "foo" and value "fooValue": variables 1`] = `{}`;

exports[`Variables setters set("foo", "fooValue")) should set variable with key "foo" and value "fooValue": variables 2`] = `
{
  "foo": {
    "type": "string",
    "value": "fooValue",
    "valueInfo": {},
  },
}
`;

exports[`Variables setters setAll(someValues)  should add someValues to variables: dirty variables 1`] = `{}`;

exports[`Variables setters setAll(someValues)  should add someValues to variables: dirty variables 2`] = `
{
  "bar": {
    "type": "integer",
    "value": 2,
    "valueInfo": {},
  },
  "foo": {
    "type": "string",
    "value": "FooValue",
    "valueInfo": {},
  },
}
`;

exports[`Variables setters setAll(someValues)  should add someValues to variables: variables 1`] = `{}`;

exports[`Variables setters setAll(someValues)  should add someValues to variables: variables 2`] = `
{
  "bar": {
    "type": "integer",
    "value": 2,
    "valueInfo": {},
  },
  "foo": {
    "type": "string",
    "value": "FooValue",
    "valueInfo": {},
  },
}
`;

exports[`Variables setters setAllTyped(someTypedValues) should add someTypedValues to variables: dirty variables 1`] = `{}`;

exports[`Variables setters setAllTyped(someTypedValues) should add someTypedValues to variables: dirty variables 2`] = `
{
  "bar": {
    "type": "string",
    "value": "barValue",
    "valueInfo": {},
  },
  "foo": {
    "type": "string",
    "value": "fooValue",
    "valueInfo": {},
  },
}
`;

exports[`Variables setters setAllTyped(someTypedValues) should add someTypedValues to variables: variables 1`] = `{}`;

exports[`Variables setters setAllTyped(someTypedValues) should add someTypedValues to variables: variables 2`] = `
{
  "bar": {
    "type": "string",
    "value": "barValue",
    "valueInfo": {},
  },
  "foo": {
    "type": "string",
    "value": "fooValue",
    "valueInfo": {},
  },
}
`;

exports[`Variables setters setTransient("fooTransient", "fooValueTransient")) should set variable with key "fooTransient" and value "fooValueTransient" and transient "true": dirty variables 1`] = `{}`;

exports[`Variables setters setTransient("fooTransient", "fooValueTransient")) should set variable with key "fooTransient" and value "fooValueTransient" and transient "true": dirty variables 2`] = `
{
  "fooTransient": {
    "type": "string",
    "value": "fooValueTransient",
    "valueInfo": {
      "transient": true,
    },
  },
}
`;

exports[`Variables setters setTransient("fooTransient", "fooValueTransient")) should set variable with key "fooTransient" and value "fooValueTransient" and transient "true": variables 1`] = `{}`;

exports[`Variables setters setTransient("fooTransient", "fooValueTransient")) should set variable with key "fooTransient" and value "fooValueTransient" and transient "true": variables 2`] = `
{
  "fooTransient": {
    "type": "string",
    "value": "fooValueTransient",
    "valueInfo": {
      "transient": true,
    },
  },
}
`;

exports[`Variables setters setTyped("baz",someTypeValue) should set typed value with key "baz": dirty variables 1`] = `{}`;

exports[`Variables setters setTyped("baz",someTypeValue) should set typed value with key "baz": dirty variables 2`] = `
{
  "baz": {
    "type": "json",
    "value": "{"name":"bazname"}",
    "valueInfo": {},
  },
}
`;

exports[`Variables setters setTyped("baz",someTypeValue) should set typed value with key "baz": variables 1`] = `{}`;

exports[`Variables setters setTyped("baz",someTypeValue) should set typed value with key "baz": variables 2`] = `
{
  "baz": {
    "type": "json",
    "value": {
      "name": "bazname",
    },
    "valueInfo": {},
  },
}
`;
